### PR TITLE
9785 - prepare_encryption and infrastructure_encryption_enabled

### PR DESCRIPTION
**Pull request by istairbn**
Initial groundwork for completion of #9785 

Two booleans missing from custom paramters: PrepareEncryption (turning on the managed identity) and RequireInfrastructureEncryption. infrastructure_encryption_enabled seemed to be the common term used elsewhere in the provider.

### CODE REVIEW

**Comment by istairbn** [comment]
@WodansSon - just seen you're doing a massive API upgrade. Is this likely to become redundant?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ ENHANCEMENTS:
 * dependencies: updating to `v0.11.19` of `github.com/Azure/go-autorest/autorest` [GH-12209]
 * dependencies: updating to `v0.9.14` of `github.com/Azure/go-autorest/autorest/adal` [GH-12209]
 * dependencies: updating the embedded SDK for Eventhub Namespaces to use API Version `2021-01-01-preview` [GH-12290]
+* `azurerm_databricks_workspace` - support for `infrastructure_encryption_enabled` and `prepare_encryption` ([#9785](https://github.com/terraform-providers/terraform-provider-azurerm/issues/9785))
```
Comment (jackofallops) [inline CHANGELOG.md:10]: Hi @istairbn  - Could you remove this from the PR please? The changelog is updated by maintainers at the point PR's are merged so they correctly reflect releases.
Thanks!

Comment (istairbn) [reply CHANGELOG.md]: @jackofallops my bad - makes sense. 
PR updated to strike the CHANGELOG changes

**Comment by jackofallops** [comment]
Hi @istairbn - I think this change is covered by #12331 which, while opened later, also includes the customer managed key part of the requirement also, so I thank you for your efforts, but I'm going to close this in favour of that one.

**Pull request closed without merging**